src/os/JournalingObjectStore.cc

```diff
@@ -191,29 +193,35 @@ bool JournalingObjectStore::ApplyManager::commit_start()
   {
     Mutex::Locker l(apply_lock);
     dout(10) << "commit_start max_applied_seq " << max_applied_seq
-	     << ", open_ops " << open_ops
-	     << dendl;
+	     << ", open_ops " << open_ops << dendl;
+  
     blocked = true;
-    while (open_ops > 0) {
-      dout(10) << "commit_start waiting for " << open_ops << " open ops to drain" << dendl;
-      blocked_cond.Wait(apply_lock);
-    }
-    assert(open_ops == 0);
-    dout(10) << "commit_start blocked, all open_ops have completed" << dendl;
     {
       Mutex::Locker l(com_lock);
-      if (max_applied_seq == committed_seq) {
-	dout(10) << "commit_start nothing to do" << dendl;
+
+      set<uint64_t>::iterator it = applying_seq_set.begin();
+      if (it == applying_seq_set.end() ){
+       	_committing_seq = committing_seq = max_applied_seq; 
+		
+      dout(10) << "commit_start committing from end: " << committing_seq 
+		<< ",current applying_seq_set size:"<< applying_seq_set.size()
+		<< ", still blocked" << dendl;
+		assert( applying_seq_set.size() == 0 );
+      }else
+      {
+        _committing_seq = committing_seq = *it - 1;
```

Comment:
@yuyuyu101 
how to track this just use on uint64_t ? 
for example:
current apply ops in flight is: 7, 8, 9, 10
when sync,  the current applying ops that has completed is  9,10
the min_applied_seq is 9, but it can't sync the seq 9, because the ops of  7, 8 don't completed.

do you have another good way to just use one uint64_t to track ? Right now, i have no idea of a way that does work.